test/e2e/e2e_suite_test.go

```diff
@@ -401,3 +407,101 @@ func tearDown(bootstrapClusterProvider bootstrap.ClusterProvider, bootstrapClust
 		bootstrapClusterProvider.Dispose(context.TODO())
 	}
 }
+
+// resolveKubernetesVersions looks at Kubernetes versions set as variables in the e2e config and sets them to a valid k8s version
+// that has an existing capi offer image available. For example, if the version is "stable-1.22", the function will set it to the latest 1.22 version that has a published reference image.
+func resolveKubernetesVersions(config *clusterctl.E2EConfig) {
+	skus := getAllCAPIImageSkus(context.TODO(), os.Getenv(AzureLocation))
+	versions := parseImageSkuNames(skus)
+
+	if config.HasVariable(capi_e2e.KubernetesVersion) {
+		resolveKubernetesVersion(config, versions, capi_e2e.KubernetesVersion)
+	}
+	if config.HasVariable(capi_e2e.KubernetesVersionUpgradeFrom) {
+		resolveKubernetesVersion(config, versions, capi_e2e.KubernetesVersionUpgradeFrom)
+	}
+	if config.HasVariable(capi_e2e.KubernetesVersionUpgradeTo) {
+		resolveKubernetesVersion(config, versions, capi_e2e.KubernetesVersionUpgradeTo)
+	}
+	if config.HasVariable(capi_e2e.KubernetesVersionManagement) {
+		resolveKubernetesVersion(config, versions, capi_e2e.KubernetesVersionManagement)
+	}
+	if config.HasVariable(AKSKubernetesVersion) {
```

Comment:
/hold

need to remove this. AKS version covered in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/aks.go#L254-L257

Reply:
/hold cancel